tests/validations/mta/mtaValidations.spec.ts

```diff
@@ -0,0 +1,284 @@
+        .returns(mockFileSystemWatcher);
+      const onDidChangeSpy = sinon.stub(mockFileSystemWatcher, "onDidChange");
+      const onDidCreateSpy = sinon.stub(mockFileSystemWatcher, "onDidCreate");
+      const onDidDeleteSpy = sinon.stub(mockFileSystemWatcher, "onDidDelete");
+
+      watchMtaYamlAndDevExtFiles(disposables);
+
+      // createFileSystemWatcher
+      expect(createFileSystemWatcherSpy.callCount).to.equal(1);
+      expect(
+        createFileSystemWatcherSpy.withArgs(`**/{mta.yaml,dev.mtaext}`).called
+      ).to.be.true;
+
+      // onDidChange
+      expect(onDidChangeSpy.callCount).to.equal(1);
+      expect(
+        onDidChangeSpy.withArgs(sinon.match.func, undefined, disposables).called
+      ).to.be.true;
+
+      // onDidCreate
+      expect(onDidCreateSpy.callCount).to.equal(1);
+      expect(
+        onDidCreateSpy.withArgs(sinon.match.func, undefined, disposables).called
+      ).to.be.true;
+
+      // onDidDelete
+      expect(onDidDeleteSpy.callCount).to.equal(1);
+      expect(
+        onDidDeleteSpy.withArgs(sinon.match.func, undefined, disposables).called
+      ).to.be.true;
+    });
+
+    it("registers to workspace.onDidChangeWorkspaceFolders with the correct parameters", async () => {
+      const onDidChangeWorkspaceFoldersSpy = sinon.stub(
+        testVscode.workspace,
+        "onDidChangeWorkspaceFolders"
+      );
+
+      watchMtaYamlAndDevExtFiles(disposables);
+
+      expect(onDidChangeWorkspaceFoldersSpy.callCount).to.equal(1);
+      expect(
+        onDidChangeWorkspaceFoldersSpy.withArgs(
+          sinon.match.func,
+          undefined,
+          disposables
+        ).called
+      ).to.be.true;
+    });
+  });
+
+  describe("updateMtaDiagnostics", async () => {
+    it("returns no diagnostics when mta.yaml does not exist", async () => {
+      await testUpdateMtaDiagnostics([]);
+    });
+
+    it("returns no diagnostics when mta.yaml and dev.mtaext have no errors", async () => {
+      sinon.stub(fsExtra, "pathExists").resolves(true);
+      sinon.stub(Mta.prototype, "validate").resolves({});
+
+      await testUpdateMtaDiagnostics([]);
+    });
+
+    it("returns diagnostics when mta.yaml and dev.mtaext have errors", async () => {
+      const expectedEntries: [
+        Uri,
+        Diagnostic[] | undefined
+      ][] = prepDiagnostics();
+      await testUpdateMtaDiagnostics(expectedEntries);
+    });
+
+    async function testUpdateMtaDiagnostics(
+      expectedCollectionEntries: [Uri, Diagnostic[] | undefined][]
+    ) {
+      const fileName = "mtaValidations.spec.js";
+      const uriPath = resolve(__dirname, fileName);
+      const uri = {
+        fsPath: uriPath,
+      } as Uri;
+
+      let collectionEntries:
+        | [Uri, Diagnostic[] | undefined][]
+        | undefined = undefined;
+      sinon.stub(testVscode.languages, "createDiagnosticCollection").returns({
+        forEach: () => {
+          return;
+        },
```

Comment:
I don't want to actually implement vscode APIs, but I think in this case if you want to check the `clear` logic it's necessary.
```suggestion
        forEach: (func: (uri: Uri, diagnostics: Diagnostic[] | undefined)) => {
          forEach(collectionEntries, (entry) => func(entry[0], entry[1]));
        },
```
(import `forEach` from lodash)